SingularityService/src/main/java/com/hubspot/singularity/data/RequestManager.java

```diff
@@ -269,4 +278,27 @@ public void deleteRequest(SingularityRequest request, Optional<String> user) {
     delete(getRequestPath(request.getId()));
   }
 
+  public List<String> getLBCleanupRequestIds() {
+    return getChildren(LB_CLEANUP_PATH_ROOT);
+  }
+
+  private String getLBCleanupPath(String requestId) {
+    return ZKPaths.makePath(LB_CLEANUP_PATH_ROOT, requestId);
+  }
+
+  public SingularityCreateResult createLBCleanupRequest(String requestId) {
+    return create(getLBCleanupPath(requestId));
+  }
+
+  public SingularityDeleteResult deleteLBCleanupRequest(String requestId) {
+    return delete(getLBCleanupPath(requestId));
+  }
+
+  public Optional<SingularityLoadBalancerUpdate> getLoadBalancerState(String requestId) {
```

Comment:
From what I interpreted it would return an Optional.absent, which would tell the cleanup method that no request has been sent yet and the cleanup still needs to be started